src/internal.c

```diff
@@ -42506,6 +42495,34 @@ int wolfSSL_TestAppleNativeCertValidation_AppendCA(WOLFSSL_CTX* ctx,
 
 #endif /* defined(__APPLE__) && defined(WOLFSSL_SYS_CA_CERTS) */
 
+#if defined(WOLFSSL_CHECK_ALERT_ON_ERR)
```

Comment:
Looks like this option is only enabled with the compatibility layer or --enable-harden-tls. Should it be enabled in more cases? Does the ssl.c description of this macro need updated?

```
* WOLFSSL_CHECK_ALERT_ON_ERR:
 *     Check for alerts during the handshake in the event of an error.
```